### PR TITLE
Feature/quiz paginator

### PR DESCRIPTION
**Pull request by alvarogonzalezDEV**

### CODE REVIEW

File: src/components/question-container.html [hunk 1]
```diff
@@ -30,6 +31,7 @@
       }
 
       .question-container {
+        width: 40%;
```
Comment (KGomes27) [inline src/components/question-container.html]: Two widths?

File: src/question-view.html [hunk 2]
```diff
@@ -43,27 +43,18 @@
...
-      data="{{routeData}}">
+    <app-route route="{{route}}" pattern="/:id" data="{{routeData}}">
     </app-route>
 
-    <quiz-status-data
-      id="quizStatus"
-      total-questions="[[quizData.questions.length]]"
-      progress="{{quizProgress}}"
-      answered-questions="{{answeredQuestions}}"
+    <quiz-status-data id="quizStatus" total-questions="[[quizData.questions.length]]" progress="{{quizProgress}}" answered-questions="{{answeredQuestions}}"
```
Comment (KGomes27) [inline src/question-view.html]: When the tag has more than two or three attributes I think it looks cleaner if we put them in separate lines.

File: src/question-view.html [hunk 3]
```diff
@@ -43,27 +43,18 @@
...
       answered-questions-count="{{answeredQuestionsCount}}">
     </quiz-status-data>
 
-    <question-tabs
-      questions="[[quizData.questions]]"
-      answered-questions="[[answeredQuestions]]"
-      selected="{{routeData.id}}">
+    <question-tabs questions="[[quizData.questions]]" answered-questions="[[answeredQuestions]]" selected="{{routeData.id}}">
```
Comment (KGomes27) [inline src/question-view.html]: Same here.

File: src/question-view.html [hunk 4]
```diff
@@ -43,27 +43,18 @@
...
     </question-tabs>
 
-    <question-container question="[[question]]" selected-answer="{{selectedAnswer}}"></question-container>
+    <question-container quiz-data="[[quizData]]" current-question-id="[[routeData.id]]" answered-questions-count="{{answeredQuestionsCount}}" question="[[question]]" selected-answer="{{selectedAnswer}}"></question-container>
```
Comment (KGomes27) [inline src/question-view.html]: And here haha

File: src/question-view.html [hunk 5]
```diff
@@ -108,14 +100,44 @@
         let question = this.quizData.questions.find((q) => {
           return Number(q.id) === Number(id);
         });
-        
+
+      //  this._setButtonDisabled(this.quizData.questions.findIndex(i => i.id == question.id));
```
Comment (KGomes27) [inline src/question-view.html]: Can we remove this comment?

File: src/components/quiz-paginator.html [hunk 6]
```diff
@@ -0,0 +1,86 @@
+<link rel="import" href="../../bower_components/polymer/polymer-element.html">
+<link rel="import" href="../styles/shared-styles.html">
+
+<dom-module id="quiz-paginator">
+  <template>
+    <style include="shared-styles">
+      :host {
+        display: flex;
+
```
Comment (KGomes27) [inline src/components/quiz-paginator.html]: trailing whitespace

File: src/components/quiz-paginator.html [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      }
+
+      .finish {
+        flex-grow: 1;
+        text-align: left;
+      }
+
+      button {
+        font-size: var(--sopra-font-xxxl);
```
Comment (KGomes27) [inline src/components/quiz-paginator.html]: Maybe we could use --sopra-font-xxl

File: src/components/question-container.html (hunk 1 above)
Comment (edu-aguilar) [reply src/components/question-container.html]: xDDDDDDDD

File: src/components/quiz-paginator.html (hunk 7 above)
Comment (edu-aguilar) [reply src/components/quiz-paginator.html]: Alvaro always with 1 size more, u know. He loves accessibility

File: src/components/quiz-paginator.html [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        margin-top: 2em;
+        border-width: 0;
+        outline: none;
+        color: var(--sopra-blue-500);
+        cursor: pointer;
+        background-color: transparent;
```
Comment (edu-aguilar) [inline src/components/quiz-paginator.html]: Must we try to do a general app-buttons.html to have there common styles to our buttons and use it through shared styles?

File: src/components/quiz-paginator.html [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+      }
+
+      button[disabled]:active {
+        opacity: 0.4;
+        cursor: not-allowed;
+        pointer-events: none;
```
Comment (edu-aguilar) [inline src/components/quiz-paginator.html]: I think this is not necessary. Button tag automatically applies disabled styles when property disabled is already set

File: src/components/quiz-paginator.html [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+        return {
+          answeredQuestionsCount: {
+            type: Number,
+            observer: '_observerIfFinished'
+          },
+          currentQuestionId: {
+            type: Number,
+            observer: '_observerRouterId'
+          },
+          quizData: Object,
```
Comment (edu-aguilar) [inline src/components/quiz-paginator.html]: you have quizData Object as component's property, that's ok, but in your comp, you only access that property to get the length. I think you must have a property quizLength(Number) instead.

File: src/components/quiz-paginator.html [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+      }
+
+      _observerIfFinished() {
+          this.finishDisabled = !(this.answeredQuestionsCount === this.quizData.questions.length)
+      }
+
+      _observerRouterId() {
+        const index = parseInt(this.currentQuestionId)
+        this.nextDisabled = index === this.quizData.questions.length ? true : false;
+        this.backDisabled = index === 1 ? true : false;
```
Comment (edu-aguilar) [inline src/components/quiz-paginator.html]: this ternaries are bad. Read both again, you only need to do that:

- this.nextDisabled = index === this.quizData.questions.length;
- this.backDisabled = index === 1;

File: src/question-view.html (hunk 4 above)
Comment (edu-aguilar) [reply src/question-view.html]: ay lmaooo

File: src/question-view.html (hunk 5 above)
Comment (edu-aguilar) [reply src/question-view.html]: ofc please, dont commit with comments even you write something important.

File: src/question-view.html [hunk 12]
```diff
@@ -108,14 +100,44 @@
...
           this.set('question', question);
           this.selectedAnswer = this.answeredQuestions[question.id] ? this.answeredQuestions[question.id] : '';
+
         }
       }
+
+      _pageChange(event) {
+
+        let data = event.detail.operation;
+        let index = this.quizData.questions.findIndex(i => i.id == this.question.id);
```
Comment (edu-aguilar) [inline src/question-view.html]: can we use here === instead == ?

File: src/question-view.html [hunk 13]
```diff
@@ -108,14 +100,44 @@
...
+          case 'finish':
+            var event = new CustomEvent('onRouteChange', { detail: { newRoute: 'result' }, composed: true });
+            this.dispatchEvent(event);
+            break;
+          default:
+            break;
+        }  
+
+        this.routeData.id =  this.quizData.questions[index].id;
+        this.notifyPath('routeData.id');
```
Comment (edu-aguilar) [inline src/question-view.html]: this is ok, but i remember you that you can do the same easily:

this.set('routeData.id', this.quizData.questions[index].id);

Remember that using polymer methods (like set) to mutate properties you don't need to use notifyPath =)

File: src/styles/app-form-items.html [hunk 14]
```diff
@@ -14,7 +14,7 @@
     background-color: var(--sopra-orange-500);
   }
 
-  .button-quiz:disabled {
+  button:disabled {
     opacity: 0.4;
   }
```
Comment (edu-aguilar) [inline src/styles/app-form-items.html]: We need to talk about this plz

File: src/components/quiz-paginator.html (hunk 9 above)
Comment (KGomes27) [reply src/components/quiz-paginator.html]: We should check if the default disabled styles are what we want and if they are, we can delete this :D

File: src/components/quiz-paginator.html (hunk 11 above)
Comment (alvarogonzalezDEV) [reply src/components/quiz-paginator.html]: omg, totaly true

File: src/components/question-container.html [hunk 15]
```diff
@@ -50,6 +51,7 @@
           <answer-item id="[[option.id]]" option="[[option]]"></answer-item>
         </template>
       </iron-selector>
+      <quiz-paginator quiz-data="[[quizData]]" current-question-id="[[currentQuestionId]]" answered-questions-count="{{answeredQuestionsCount}}"></quiz-paginator>
```
Comment (KGomes27) [inline src/components/question-container.html]: Let's put this in separate lines hahaha

File: src/components/quiz-paginator.html (hunk 10 above)
Comment (KGomes27) [reply src/components/quiz-paginator.html]: You missed this comment @alvarogonzalezDEV

File: src/components/quiz-paginator.html (hunk 6 above)
Comment (KGomes27) [reply src/components/quiz-paginator.html]: You missed this comment @alvarogonzalezDEV

File: src/question-view.html (hunk 13 above)
Comment (KGomes27) [reply src/question-view.html]: Try this and see if it works @alvarogonzalezDEV :)

File: src/components/question-container.html [hunk 16]
```diff
@@ -68,7 +74,10 @@
             type: String,
             value: '',
             notify: true
-          }
+          },
+            answeredQuestionsCount: Number,
+            quizData: Object,
```
Comment (KGomes27) [inline src/components/question-container.html]: Are we changing this to quizDataLength?

File: src/components/quiz-paginator.html (hunk 8 above)
Comment (edu-aguilar) [reply src/components/quiz-paginator.html]: please add mixins to your CSS class

File: src/components/quiz-paginator.html [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+          nextDisabled: {
+            type: Boolean,
+            value: false
+          }
+        }
+      }
+
+      _changeRoute(event) {
+        
+        debugger;
```
Comment (edu-aguilar) [inline src/components/quiz-paginator.html]: remove this plzzz

File: src/components/quiz-paginator.html [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+        let data = event.target.getAttribute('data');
+        var event = new CustomEvent('onPageQuestionChange', { detail: { operation: data }, composed: true });
```
Comment (edu-aguilar) [inline src/components/quiz-paginator.html]: use const instead var

File: src/styles/app-form-items.html (hunk 14 above)
Comment (edu-aguilar) [reply src/styles/app-form-items.html]: disabled is a native property from buttons. This is not needed

**Pull request merged**